app/presenters/queries/content_item_presenter.rb

```diff
@@ -12,43 +12,62 @@ def self.present(content_item)
         present_many(content_items).first
       end
 
-      def self.present_many(content_item_scope, fields: nil, order: { public_updated_at: :desc }, offset: 0, limit: nil)
-        presenter = new(content_item_scope, fields)
-        presenter.order = order
-        presenter.limit = limit
-        presenter.offset = offset
+      def self.present_many(content_item_scope, fields: nil, order: { public_updated_at: :desc }, offset: 0, limit: nil, locale: nil)
+        presenter = new(content_item_scope, fields: fields, order: order, limit: limit, offset: offset, locale: locale)
         presenter.present
       end
 
-      def initialize(content_item_scope, fields = nil)
+      def initialize(content_item_scope, fields: nil, order: {}, limit: nil, offset: 0, locale: nil)
         self.content_item_scope = content_item_scope
         self.fields = fields
+        self.order = order
+        self.limit = limit
+        self.offset = offset
+        self.locale = locale
       end
 
       def present
         group_items(groups).compact
       end
 
+      def total
+        ActiveRecord::Base.connection.execute(total_query).first["count"].to_i
+      end
+
     private
 
-      attr_accessor :content_item_scope, :fields
+      attr_accessor :content_item_scope, :fields, :locale
 
-      def groups
-        scope = join_supporting_objects(content_item_scope)
-        scope = select_fields(scope).order(order)
+      def total_query
+        sql = content_item_scope.select(:content_id)
+        sql = sql.group(:content_id) unless locale == "all"
+        "SELECT COUNT(*) FROM (#{sql.to_sql}) total"
```

Comment:
Some local testing shows that this is pretty slow on document_types with a decent amount of data, eg cma_cases, but can be massively speeded up (70ms vs 1700ms) by adding an index to translations on (content_item_id, locale). Can you add a migration?